### PR TITLE
Implement userProfileController

### PR DESCRIPTION
**Pull request by jakji93**
- Implement UserProfileController
- adding some additional documentation on router

Note:
- GetAllProfile might a security concern, don't think we should give access to all profiles
- Delete might not need, should we have profile live for ever?

### CODE REVIEW

**Review by ckev07 — APPROVED** [review]

File: server/src/controllers/userProfileController.js [hunk 1]
```diff
@@ -1,126 +1,149 @@
...
  * @desc    creturns a list of all the user profiles
  * @route   GET /userProfile
  * @access  Private
  */
-const getAllUserProfiles = (req, res) => {
-  const allProfiles = Object.values(userProfiles);
 
-  res.json(allProfiles);
-};
+// Dont' think we should give access to all user profiles
```
Comment (kennyjhcheng) [inline server/src/controllers/userProfileController.js:10]: yea I agree, if we don't use this route, we should remove it

File: server/src/routes/UserProfile.js [hunk 2]
```diff
@@ -55,87 +55,91 @@ const router = express.Router();
...
  *        ...string
  *    ],
- *    "dietList": [
+ *    "dietRestriction": [
  *        ..string
  *    ]
  *  }
  */
-router.get('/:userID', getUserProfileById);
+router.get('/', getUserProfileById);
```
Comment (kennyjhcheng) [inline server/src/routes/UserProfile.js:104]: have to add the middleware `protect` from `auth.js` here to get access to `req.user._id`

File: server/src/controllers/userProfileController.js [hunk 3]
```diff
@@ -1,126 +1,149 @@
...
-  const userProfile = userProfiles[userID];
-
-  if (userProfile) {
-    res.json(userProfile);
-  } else {
-    res.status(404).json({ error: 'User profile not found' });
+const getUserProfileById = asyncHandler(async (req, res) => {
+  const userProfile = await UserProfile.findOne({ userInfoID: req.user._id });
+  if (!userProfile) {
+    res.status(400);
```
Comment (kennyjhcheng) [inline server/src/controllers/userProfileController.js:28]: Add a `.json({ message: '...' });`

File: server/src/controllers/userProfileController.js (hunk 1 above)
Comment (jakji93) [reply server/src/controllers/userProfileController.js:10]: Same for delete, do we allow for deletion?

File: server/src/controllers/userProfileController.js [hunk 4]
```diff
@@ -1,129 +1,141 @@
...
+      || !userProfile.experience
+      || !userProfile.goals
+      || !userProfile.apiKey
+      || !userProfile.firstName
+      || !userProfile.lastName) {
+    res.status(400).json({ message: 'Please include all required fields' });
+    throw new Error('Please include all required fields');
+  }
+  userProfile.userInfoID = req.user._id;
+  const newProfile = await UserProfile.findOneAndUpdate(
```
Comment (kennyjhcheng) [inline server/src/controllers/userProfileController.js:93]: `updatedProfile` is a better name for this const

File: server/src/controllers/userProfileController.js (hunk 1 above)
Comment (kennyjhcheng) [reply server/src/controllers/userProfileController.js]: Yes, I think we can allow them to delete their profile.

Comment (kennyjhcheng) [reply server/src/controllers/userProfileController.js]: but we have to not only delete their Profile, but also delete their `UserInfo` and `FitnessPlan`.

So maybe it should be delete for `UserInfo` would delete all the records for the user in all collections

Comment (jakji93) [reply server/src/controllers/userProfileController.js]: I guess I can leave the implemented delete for now and decide on it later? I think have it in UserInfo would be better so it deletes all as a bundle

**Review by kennyjhcheng — APPROVED** [review]

**Review by ckev07 — APPROVED** [review]

**Pull request merged**